### ISSUE TITLE
[Feature] Add a back button to improve HTML Report & Trace viewer in iframe context

### ISSUE THREAD

**Issue by Az8th** [microsoft/playwright#28389]
Hello,

This should be a quick one 😄

__TL;DR:__
Could you please add a 'back' button to the HTML report ? It would help for easier navigation, especially when it is integrated in an iframe.

__My oddly specific usecase (aka the longer version):__
I use Azure DevOps to manage and track backlog items and test runs, but it lacks a lot of features concerning tests. which I try to add by using the dashboard menu. I fullfilled most of my needs by adding widgets, one being a webpage viewer, serving the last html report produced by the CI in an iframe, which defaults to failed runs.

It works nicely, I can search, see the details of a specific test or attached traces/videos/screenshots, but I'm then stuck to those subviews. Normally I would use backspace or the browser button to go back, but it can't apply in an iframe context. Using right click > previous works though, but it is not very convenient. A simple button could fix that easily, and may even improve UX in a classic context ;)

Also, there is no scrollbar in this context, which can be very unconvenient when there is multiple tests (middle click doesn't works here). I know one can be added using css, but I have no control over it. Is there any way its presence could be forced on the reporter side ?

Thanks ❤️

**Comment by pavelfeldman** [comment]
The fact that history and scrollbars do not work suggest that there is something wrong with your embedder. Browsers support history inside iframes, so you should be able to click "back" w/o problems. I guess in this case your embedder intercepts history events and performs its own actions. Same goes for the scroll bars - if the content of the iframe is larger than the iframe, it'll get the scroll bars.

Working around the browser history bug in the embedder might be hard - imagine we open an image, not a trace from the html report; there will be no back button.

**Comment by Az8th** [comment]
Effictively, that's whatI was meaning by I don't have control over the CSS, and why I was asking for the add of a back button. 

Not sure if I understood your example with image, but using an icon (like left to the search bar) redirecting to `javascript:history.back()` be enough ? It could even use script to be written in the dom only if `history.length > 1`

**Comment by pavelfeldman** [comment]
I am not sure we are on the same page. What I was saying is that Back button in the browser works in the iframe, so there is nothing for us to fix here. We don't want to be mimicking browser functionality in-page.

**Comment by Az8th** [comment]
Okay, I will close this as this is not in scope then. Thanks for your help though !
As a workaround, I will produce my own iframe integration plugin for Azure DevOps
